frame/society/src/lib.rs

```diff
@@ -1014,61 +1073,64 @@ pub mod pallet {
+		/// - `max_strikes`: The maximum number of strikes a member may get before they become
+		///   suspended and may only be reinstated by the founder.
+		/// - `candidate_deposit`: The deposit required to make a bid for membership of the group.
 		/// - `rules` - The rules of this society concerning membership.
 		///
-		/// # <weight>
-		/// - Two storage mutates to set `Head` and `Founder`. O(1)
-		/// - One storage write to add the first member to society. O(1)
-		/// - One event.
-		///
-		/// Total Complexity: O(1)
-		/// # </weight>
+		/// Complexity: O(1)
 		#[pallet::weight(T::BlockWeights::get().max_block / 10)]
-		pub fn found(
+		pub fn found_society(
 			origin: OriginFor<T>,
 			founder: T::AccountId,
 			max_members: u32,
+			max_intake: u32,
+			max_strikes: u32,
+			candidate_deposit: BalanceOf<T, I>,
 			rules: Vec<u8>,
 		) -> DispatchResult {
 			T::FounderSetOrigin::ensure_origin(origin)?;
-			ensure!(!<Head<T, I>>::exists(), Error::<T, I>::AlreadyFounded);
+			ensure!(!Head::<T, I>::exists(), Error::<T, I>::AlreadyFounded);
 			ensure!(max_members > 1, Error::<T, I>::MaxMembers);
 			// This should never fail in the context of this function...
-			<MaxMembers<T, I>>::put(max_members);
-			Self::add_member(&founder)?;
-			<Head<T, I>>::put(&founder);
-			<Founder<T, I>>::put(&founder);
+			let params = GroupParams { max_members, max_intake, max_strikes, candidate_deposit };
+			Parameters::<T, I>::put(params);
+			Self::insert_member(&founder, 1)?;
+			Head::<T, I>::put(&founder);
+			Founder::<T, I>::put(&founder);
 			Rules::<T, I>::put(T::Hashing::hash(&rules));
 			Self::deposit_event(Event::<T, I>::Founded { founder });
 			Ok(())
 		}
 
-		/// Annul the founding of the society.
+		/// Dissolve the society and remove all members.
 		///
 		/// The dispatch origin for this call must be Signed, and the signing account must be both
 		/// the `Founder` and the `Head`. This implies that it may only be done when there is one
 		/// member.
 		///
-		/// # <weight>
-		/// - Two storage reads O(1).
-		/// - Four storage removals O(1).
-		/// - One event.
-		///
 		/// Total Complexity: O(1)
-		/// # </weight>
 		#[pallet::weight(T::BlockWeights::get().max_block / 10)]
-		pub fn unfound(origin: OriginFor<T>) -> DispatchResult {
+		pub fn dissolve(origin: OriginFor<T>) -> DispatchResult {
 			let founder = ensure_signed(origin)?;
-			ensure!(Founder::<T, I>::get() == Some(founder.clone()), Error::<T, I>::NotFounder);
-			ensure!(Head::<T, I>::get() == Some(founder.clone()), Error::<T, I>::NotHead);
+			ensure!(Founder::<T, I>::get().as_ref() == Some(&founder), Error::<T, I>::NotFounder);
+			ensure!(MemberCount::<T, I>::get() == 1, Error::<T, I>::NotHead);
 
-			Members::<T, I>::kill();
+			#[allow(deprecated)]
+			Members::<T, I>::remove_all(None);
+			MemberCount::<T, I>::kill();
+			#[allow(deprecated)]
+			MemberByIndex::<T, I>::remove_all(None);
+			#[allow(deprecated)]
+			Members::<T, I>::remove_all(None);
```

Comment:
Besides the 2 `Members::<T, I>::remove_all(None)`, what about the other Storage entries (eg. `Parameters`, `Bids`, etc)?

We'd also have funds in `Pot` (maybe in `Payouts` too), should we move them to somewhere or let them there to a future new Society?

Reply:
Let them there for the new version.